app/src/ui/lib/application-theme.ts

```diff
@@ -54,3 +54,26 @@ export function getPersistedThemeName(): string {
 export function setPersistedTheme(theme: ApplicationTheme) {
   localStorage.setItem(applicationThemeKey, getThemeName(theme))
 }
+
+// The key under which the decision to automatically switch the theme is persisted
+// in localStorage.
+const automaticallySwitchApplicationThemeKey = 'autoSwitchTheme'
+
+/**
+ * Load the whether or not the user wishes to automatically switch the selected theme from the persistent
+ * store (localStorage). If no theme is selected the default
+ * theme will be returned.
+ */
+export function getAutoSwitchPersistedTheme(): boolean {
+  return localStorage.getItem(automaticallySwitchApplicationThemeKey) === 'true'
```

Comment:
We have `getBoolean` and `setBoolean` now that should make some of this plumbing unneccesary - see #5979 for more info